docs/guides/hugo.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Hugo
+lang: en-US
+meta:
+  - name: description
+    content: PurgeCSS can be used with Hugo via Hugo Pipes asset processing
+  - itemprop: description
+    content: PurgeCSS can be used with Hugo via Hugo Pipes asset processing
+  - property: og:url
+    content:  https://purgecss.com/guides/hugo
+  - property: og:site_name
+    content: purgecss.com
+  - property: og:type
+    content: website
+  - property: og:image
+    content: https://i.imgur.com/UEiUiJ0.png
+  - property: og:locale
+    content: en_US
+  - property: og:title
+    content: Remove unused CSS - PurgeCSS
+  - property: og:description
+    content: PurgeCSS can be used with Hugo via Hugo Pipes asset processing
+---
+
+# Hugo
+
+> Hugo is one of the most popular open-source static site generators. With its amazing speed and flexibility, Hugo makes building websites fun again.
+
+PurgeCSS can be used with [Hugo](https://gohugo.io/) via Hugo Pipes asset processing.
+
+## Tooling
+
+1. Install Hugo
+1. Install Node.js
+
+## Write Stats
+
+In your `config.toml` file, add this:
+
+```toml
+[build]
+  writeStats = true
+```
+
+Or, If using a `config.yaml` file, add this:
+
+```yaml
+build:
+  writeStats: true
+```
+
+This tells Hugo to write a `hugo_stats.json` file to the project root as part of the build. It includes all tags, classes, and ids from your `*.html` templates.
+
+## Node Packages
+
+Run this to install the necessary packages:
+
+```
+npm install postcss postcss-cli @fullhuman/postcss-purgecss
+```
+
+If the `package.json` file at the project root doesn't exist yet, this will create it.
+
+If it's not already there, add `node_modules/` to your `.gitignore` file.
+
+## PostCSS Config File
+
+Create a `postcss.config.js` file at the project root with these contents:
+
+```js
+const purgecss = require('@fullhuman/postcss-purgecss')({
+  content: ['./hugo_stats.json'],
+  defaultExtractor: content => {
+    const els = JSON.parse(content).htmlElements;
+    return [
+      ...(els.tags || []),
+      ...(els.classes || []),
+      ...(els.ids || []),
+    ];
+  },
+  safelist: []
+});
+
+module.exports = {
+  plugins: [
+    ...(process.env.HUGO_ENVIRONMENT === 'production' ? [purgecss] : [])
+  ]
+};
```

Comment:
I went through the guide and it works well! I tried it with a theme and it has some javascript in it that needs to be included in the `content` of PurgeCSS. 
You can get the hugo stats separately and combine it with PurgeCSS default extractor in the `defaultExtractor` option.
There is another form that can be returned from the extractor, with `classes`, `tags`, `ids`, ... It will work nicely with hugo_stats. Below is what I ended up with when I tried it with portio theme.

```js
const postCSSPurgeCSS = require('@fullhuman/postcss-purgecss');
const hugoStats = fs.readFileSync('./hugo_stats.json').toString();
const htmlElements = JSON.parse(hugoStats).htmlElements;
const extractFromHTMLElements = {
  tags: htmlElements.tags || [],
  classes: htmlElements.classes || [],
  ids: htmlElements.ids || [],
  undetermined: [],
  attributes: {}
}

const purgecss = postCSSPurgeCSS({
  content: ['./themes/portio/assets/js/**/*.js'],
  defaultExtractor: content => {
    const undetermined = content.match(/[A-Za-z0-9_-]+/g) || []
    return {
      ...extractFromHTMLElements,
      undetermined,
      attributes: {}
    };
  },
  safelist: []
});

module.exports = {
  plugins: [
    ...(process.env.HUGO_ENVIRONMENT === 'production' ? [purgecss] : [])
  ]
};
```

Reply:
Good to know on the alternate approach! 

My concern with the snippet is that not all hugo themes use JS. Some are just plain HTML and CSS. 

So in that case, what would the value of the `content` array be? Or could it just be an empty array if there was no JS to parse?

Reply:
Yes, `content` would be an empty array in this situation

Reply:
Got it. 

I’m going to play with this in a few days then will report back.

Reply:
Updated https://github.com/FullHuman/purgecss/pull/918/commits/b06d00e648921de75bfbf9b39cb87613276d8a7e

Reply:
Actually, after more testing, I've reverted that commit. It makes PurgeCSS too eager in its purging.

Reply:
> @Ffloriel - any issue with keeping the `postcss.config.js` code as-is?

Bump